NtupleProducer/plugins/HTauTauNtuplizer.cc

```diff
@@ -2631,89 +2657,88 @@ int HTauTauNtuplizer::FillJet(const edm::View<pat::Jet> *jets, const edm::Event&
     _bdiscr21.push_back(ijet->bDiscriminator("pfParticleNetAK4JetTags:probuds"));
     _bdiscr22.push_back(ijet->bDiscriminator("pfParticleNetAK4JetTags:probg"));
 
-    //PF jet ID
-    float NHF                 = ijet->neutralHadronEnergyFraction();
-    float NEMF                = ijet->neutralEmEnergyFraction();
-    float CHF                 = ijet->chargedHadronEnergyFraction();
-    float MUF                 = ijet->muonEnergyFraction();
-    float CEMF                = ijet->chargedEmEnergyFraction();
-    int   NumNeutralParticles = ijet->neutralMultiplicity();
-    int   chargedMult         = ijet->chargedMultiplicity();
-    int   NumConst            = ijet->chargedMultiplicity()+NumNeutralParticles;
-    float CHM                 = ijet->chargedMultiplicity();
-    float absjeta             = fabs(ijet->eta());
-
-    _jets_chEmEF .push_back(CEMF);  
-    _jets_chHEF  .push_back(CHF); 
-    _jets_nEmEF  .push_back(NEMF);
-    _jets_nHEF   .push_back(NHF);
-    _jets_chMult .push_back(chargedMult);  
-    _jets_neMult .push_back(NumNeutralParticles);  
-    _jets_MUF    .push_back(MUF);
-
-
     // JetID
     // https://twiki.cern.ch/twiki/bin/view/CMS/JetID13TeVUL
-    // Same recommendations for 2016 pre- and post- VFP
+    float NHF = ijet->neutralHadronEnergyFraction();
+    float NEMF = ijet->neutralEmEnergyFraction();
+    float CHF = ijet->chargedHadronEnergyFraction();
+    float MUF = ijet->muonEnergyFraction();
+    float CEMF = ijet->chargedEmEnergyFraction();
+    int neutralMult = ijet->neutralMultiplicity();
+    int chargedMult = ijet->chargedMultiplicity();
+    int NumConst = chargedMult+neutralMult;
+    float absjeta = fabs(ijet->eta());
+
+    _jets_chEmEF.push_back(CEMF);
+    _jets_chHEF.push_back(CHF);
+    _jets_nEmEF.push_back(NEMF);
+    _jets_nHEF.push_back(NHF);
+    _jets_chMult.push_back(chargedMult);
+    _jets_neMult.push_back(neutralMult);
+    _jets_MUF.push_back(MUF);
+
     int jetid=0;
     bool tightJetID = false;
     bool tightLepVetoJetID = false;
-    //2016 data
+
+    //2016
     if (theYear == 2016)
     {
       if (absjeta <= 2.4)
       {
-        tightJetID        = ( NHF<0.90 && NEMF<0.90 && NumConst>1 && CHF>0   && CHM>0 );
-        tightLepVetoJetID = ( NHF<0.90 && NEMF<0.90 && NumConst>1 && MUF<0.8 && CHF>0 && CHM>0 && CEMF<0.80 );
+        tightJetID = NHF<0.9 && NEMF<0.9 && NumConst>1 && CHF>0 && chargedMult>0;
+        tightLepVetoJetID = tightLepVetoJetID && MUF<0.8 && CEMF<0.8;
```

Comment:
won't this always be `false`?

Reply:
I'm guessing it should rather be:
```suggestion
        tightLepVetoJetID = tightJetID && MUF<0.8 && CEMF<0.8;
```

Reply:
Ah yes of course